### PR TITLE
add basic docker pull matrix

### PR DESCRIPTION
**Pull request by aweiteka**

### CODE REVIEW

**Comment by aweiteka** [comment]
@miabbott could you review?

File: steps/containers_kube.py [hunk 1]
```diff
@@ -9,6 +9,13 @@ def step_impl(context, image, host):
                                host,
                                module_args='docker pull %s' % image)
 
+@then('docker pull "{image}"')
+def step_impl(context, image):
```
Comment (miabbott) [inline steps/containers_kube.py:13]: Do you want to pass in `host` here to retain flexibility?

File: steps/containers_kube.py [hunk 2]
```diff
@@ -9,6 +9,13 @@ def step_impl(context, image, host):
...
+    '''docker pull image'''
+    assert context.remote_cmd('command',
+                               remote_user='root',
```
Comment (miabbott) [inline steps/containers_kube.py:15]: Does `remote_user` actually work yet?  I think we need to fix up the dynamic inventory script first.

File: features/atomic/docker_pull.feature [hunk 3]
```diff
@@ -0,0 +1,29 @@
+Feature: Docker pull
+  Run 'docker pull' on list of images
+
+  Scenario Outline: docker pull
+      Given "all" hosts from dynamic inventory
+       Then docker pull "<image>:<tag>"
+
+  Examples: docker images
+    | image     | tag    |
```
Comment (miabbott) [inline features/atomic/docker_pull.feature]: Formatting slightly off here when viewing via GitHub.  Might be correct in a checkout, though.

File: steps/containers_kube.py (hunk 2 above)
Comment (aweiteka) [inline steps/containers_kube.py:15]: `remote_user` is an ansible API arg. This is a hack until we add support for this in the dynamic inventory script. Static inventory supports this so `remote_user` wouldn't be necessary.

File: steps/containers_kube.py [hunk 4]
```diff
@@ -2,13 +2,19 @@
 
 from behave import *
 
-@when('image "{image}" is pulled on "{host}"')
-def step_impl(context, image, host):
+@when('docker pull "{image}"')
```
Comment (aweiteka) [inline steps/containers_kube.py:5]: I'm leaning towards not providing the "host" option for every step. Can we rely on "target_host" for this?

Comment (miabbott) [inline steps/containers_kube.py:5]: Yeah, it should "just work"  :)

**Pull request merged**